lib/services/notification-api-service.js

```diff
@@ -43,18 +41,46 @@ function NotificationApiServiceFactory(
 
         if (_.has(message, 'nodeId')) {
             return self.postNodeNotification(message);
-        } else if (_.has(message, 'taskId')) {
-            if (_.has(message, 'progress')) {
-                // This will be progress update notification if taskId is specified
-                return self.postProgressNotification(message.taskId, message.progress);
-            }
         }
         else {
             // This will be a broadcast notification if no id (like nodeId) is specified
             return self.postBroadcastNotification(message);
         }
     };
 
+    NotificationApiService.prototype.publishTaskProgress = function(message) {
+        if(!_.isString(message.taskId)) {
+            throw new Errors.BadRequestError('taskId is required for progress notification');
+        }
+        if(!message.maximum) {
+            throw new Errors.BadRequestError('maximum is required for progress notification');
+        }
+        if(!message.value) {
+            throw new Errors.BadRequestError('maximum is required for progress notification');
+        }
+
+        if (message.value) {
+            message.value = parseInt(message.value);
+        }
+        if (message.maximum) {
+            message.maximum = parseInt(message.maximum);
+        }
+        var progressData = _.pick(message, ['maximum', 'value', 'description']);
+
+        return waterline.taskdependencies.findOne({taskId: message.taskId})
+        .then(function(task) {
+            if (_.get(task, 'graphId') === undefined) {
+                throw new Errors.BadRequestError('Cannot find the active task for taskId ' + message.taskId); //jshint ignore: line
```

Comment:
The error message doesn't match with the real function, `waterline.taskdependencies.findOne` is only trying to find a task, this doesn't mean the returned task is the active task. The definition of active task is the task that is running (RackHD is executing its job code).